Sources/X509/Extensions.swift

```diff
@@ -79,23 +79,22 @@ extension Certificate {
     /// This interface also makes it easy to mark specific extensions as critical.
     public struct Extensions {
         @usableFromInline
-        var _extensions: [Extension]
-
-        /// Produce a new Extensions container from an array of ``Certificate/Extension``.
-        ///
-        /// - Parameter extensions: The base extensions.
-        @inlinable
-        public init(extensions: [Extension]) {
-            // TODO(cory): Police uniqueness
-            self._extensions = extensions
-        }
-
+        var _extensions: [Certificate.Extension]
+        
         /// Produce a new Extensions container from a collection of ``Certificate/Extension``.
         ///
         /// - Parameter extensions: The base extensions.
+        /// - Throws: if multiple extensions have the same OID
         @inlinable
-        public init<Elements>(_ extensions: Elements) where Elements: Sequence, Elements.Element == Extension {
+        public init<Elements>(_ extensions: Elements) throws where Elements: Sequence, Elements.Element == Extension {
             self._extensions = Array(extensions)
+            
+            var extensionOIDs = Set<ASN1ObjectIdentifier>()
```

Comment:
We should probably avoid the transient allocation of a `Set` here.

Reply:
Yup, but for small `n` it's vastly cheaper than allocating the Set and doing the hashing.